frontend/hub/api/request.ts

```diff
@@ -57,6 +59,41 @@ async function hubRequestCommon<T extends object | TaskResponse>(
   };
 }
 
+export async function hubPostRequestFile(
+  url: string,
+  file: Blob,
+  repository?: string,
+  signed_collection?: string,
+  signal?: AbortSignal
+): Promise<unknown> {
+  const body = new FormData();
+  body.append('file', file);
+  {
+    repository && body.append('repository', repository);
+  }
+  {
+    signed_collection && body.append('signed_collection', signed_collection);
+  }
```

Comment:
this looks like the same code as in crud/Data .. should we really be adding this in both places?
(Looks pretty hub specific)

Reply:
you're right, it was in crud/Data, used in uploadCollection.tsx I believe... but since it's used nowhere else, should we move it permanently down to requests? I moved it there initially because it was hub-specific

Reply:
Any code that appends repository and signed collections is not something the other projects can use.
It's not really something hub could use for other file uploads either.

So.. this is not really a `hubPostRequestFile` or a `postRequestFile`, this is more like `hubUploadCollectionSignature`.

But then again, if we could use actual functions to call apis, instead of all the would-be generic code, things would be much simpler everywhere ... so...  *shrug*, as long as you make sure other hub file uploads still work, :+1: